server/subscribe.go

```diff
@@ -22,51 +22,59 @@ const (
 	JIRA_SUBSCRIPTIONS_KEY = "jirasub"
 )
 
+type SubscriptionFilters struct {
+	Events     StringSet `json:"events"`
+	Projects   StringSet `json:"projects"`
+	IssueTypes StringSet `json:"issue_types"`
+}
+
 type ChannelSubscription struct {
 	Id        string              `json:"id"`
 	ChannelId string              `json:"channel_id"`
-	Filters   map[string][]string `json:"filters"`
+	Filters   SubscriptionFilters `json:"filters"`
 }
 
 type ChannelSubscriptions struct {
 	ById          map[string]ChannelSubscription `json:"by_id"`
-	IdByChannelId map[string][]string            `json:"id_by_channel_id"`
-	IdByEvent     map[string][]string            `json:"id_by_event"`
+	IdByChannelId map[string]StringSet           `json:"id_by_channel_id"`
+	IdByEvent     map[string]StringSet           `json:"id_by_event"`
 }
 
 func NewChannelSubscriptions() *ChannelSubscriptions {
 	return &ChannelSubscriptions{
 		ById:          map[string]ChannelSubscription{},
-		IdByChannelId: map[string][]string{},
-		IdByEvent:     map[string][]string{},
+		IdByChannelId: map[string]StringSet{},
+		IdByEvent:     map[string]StringSet{},
 	}
 }
 
+func (sub *ChannelSubscription) Events() StringSet {
```

Comment:
0/5 why have these functions that access public struct members on public structs?